app/models/miq_ae_class.rb

```diff
@@ -174,6 +175,16 @@ def self.display_name(number = 1)
     n_('Automate Class', 'Automate Classes', number)
   end
 
+  def set_fqname
+    self.domain_id ||= domain&.id || ae_namespace.domain_id
+    self.relative_path = [ae_namespace.relative_path, name].compact.join("/")
+  end
+
+  def set_children_fqname
```

Comment:
lets test this

also, do you notice the tests running longer? or do they seem the same

I think importing may become slow

maybe we should limit to the cases when the fqname on this object has changed

instead of save, maybe we could update_attribute? - push the change to the children.
This class knows if there is a change and has the parent object in memory (aka self)
so instead of asking all objects (please look up your parent and see if you've changed), we can tell them definitively that they have changed, and pass the object so they don't have to look up any more objects.